[Merged by Bors] - Fixes #2434 : Add option for `secret_name` in TLS related configs.

udsamani opened the pull request:
We add an option for `secret_name` in TLS related configs. This way the user has more fine grained control on what name of secret the user can give. This PR makes changes at all appropriate locations.

udsamani commented:
@sehz Please review when possible. Thank you :)

sehz commented on crates/fluvio-test-util/tls.rs:
```diff
@@ -7,18 +7,22 @@ pub fn cert_dir() -> PathBuf {
 
 pub fn load_tls(client_user: &str) -> (TlsPolicy, TlsPolicy) {
     const DOMAIN: &str = "fluvio.local";
+    const CLIENT_SECRET_NAME: &str = "fluvio-client-tls";
```
use `fluvio-types`

sehz reviewed (changes requested):
Generally looks good.  Just minor comments about moving the secret name to `fluvio-type` crates.  Don't forget to bump the crate version

udsamani commented:
@sehz  Have addressed the comments and also bumped the crate version. Ideally I feel that we should bump the version and not the patch version as we are changing the API. Let me know if you think otherwise, I can make changes appropriately.

sehz commented:
Sorry. Missed communication, only published crate version should be bumped up.

udsamani commented on crates/fluvio/src/config/tls.rs:
```diff
@@ -169,6 +172,8 @@ pub struct TlsPaths {
     /// Path to Certificate Authority certificate
     pub ca_cert: PathBuf,
+    /// Secret name while putting in kubernetes
+    pub secret_name: String,
```
@sehz We do need it because we use `TlsPaths` and `TlsCerts` in `fluvio-cluster` package. We use make use of `From` trait to convert between them interchangeably. Let me know if it makes sense !

udsamani replied:
Fixed. Moved it to only Kube related configs.

udsamani commented:
@sehz Am I doing something wrong while bumping 'fluvip-types' ? I bumped it from 0.3.9 to 0.4.0 ! Or should it be 0.3.10 ? I see build failure on CI because of this ! However locally it builds fine.

morenol commented on crates/fluvio-sc/src/k8/objects/spg_group.rs:
```diff
@@ -293,7 +293,7 @@ mod k8_convert {
                 name: "tls".to_owned(),
                 secret: Some(SecretVolumeSpec {
-                    secret_name: "fluvio-tls".to_owned(), // fixed
+                    secret_name: tls.secret_name.clone().unwrap(),
```
we could use unwrap_or_else(|| "tls-secret".to_string()), or change ScOpt to make this a string with a default value of "tls-secret"

udsamani replied:
Ack. Will make that change appropriately.

udsamani replied:
Fixed.

sehz commented:
> @sehz Am I doing something wrong while bumping 'fluvip-types' ? I bumped it from 0.3.9 to 0.4.0 ! Or should it be 0.3.10 ? I see build failure on CI because of this ! However locally it builds fine.

Yes, it should be bumped to `0.3.10` since this is incremental changes.

sehz reviewed (changes requested):
From looking at changes, this leaks Kubernets concern into the rest of the platform.  Can you come up with changes that safely only add to Kubernetes-related components?

udsamani commented on crates/fluvio-cluster/src/cli/start/tls.rs:
```diff
@@ -30,13 +31,21 @@ pub struct TlsOpt {
+
+    /// TLS: Secret name used for server while adding to kubernetes
+    #[clap(long)]
+    pub server_secret_name: Option<String>,
```
I see. Let me take a stab on how can I separate those concerns.

udsamani replied:
Have made necessary changes please check.

udsamani commented:
@sehz Don't think so the failure is related to my change. I see a similar failure here https://github.com/infinyon/fluvio/actions/runs/3423665089/jobs/5702625557. Looks like it is related to the ongoing issue of `spu` replication in CI.

sehz commented:
CI failure should be gone after rebasing

udsamani commented:
@sehz I did rebase. If you can approve for CI again !

udsamani commented:
@sehz Let me know if it looks good now.

morenol commented on crates/fluvio-cluster/src/cli/start/mod.rs:
```diff
@@ -101,6 +101,14 @@ pub struct K8Install {
+    tls_client_secret_name: String,
+
+    /// TLS: Server secret name while adding to Kubernetes
+    #[clap(long, default_value = "fluvio-tls")]
```
could you use `TLS_SERVER_SECRET_NAME` instead?

I think that you can use:

```rust
fn default_ tls_server_secret_name() -> &'static str {
    TLS_SERVER_SECRET_NAME
}
```

udsamani replied:
@morenol I am not sure if clap takes function values. We will have to keep it string.

morenol replied:
ohh, right I said that because I wasn't sure that 

```
    #[clap(long, default_value = TLS_SERVER_SECRET_NAME)]
```

would work

udsamani replied:
Yup. That change i did push :)

sehz reviewed (approved):
Excellent.  These are nice and clean changes.  Thanks for contribution

sehz commented:
bors r+
